feg/gateway/services/testcore/pcf/servicers/mock_driven_test.go

```diff
@@ -0,0 +1,298 @@
+	postUrl := fmt.Sprintf("http://%s%s/sm-policies/%s/update", pcfAddr, BASE_PATH, smPolicyId)
+	return http.Post(postUrl, "application/json", bytes.NewBuffer([]byte(payload)))
+}
+
+func sendDeleteSmPolicy(pcfAddr string, smPolicyId string, payload string) (*http.Response, error) {
+	postUrl := fmt.Sprintf("http://%s%s/sm-policies/%s/delete", pcfAddr, BASE_PATH, smPolicyId)
+	return http.Post(postUrl, "application/json", bytes.NewBuffer([]byte(payload)))
+}
+
+func createPolicyDecision() string {
+	policyDecisionStr := `{
+		"pccRules": {
+			"rule1": {
+				"pccRuleId": "%s",
+				"flowInfos": [{
+					"flowDescription": "permit in ip from 0.0.0.0/0 to 4.2.2.4"
+				}],
+				"precedence": 1,
+				"refQosData": ["qos_data1"],
+				"refTcData": ["tc_data1"],
+				"refChgData": ["chg_data1"],
+				"refUmData": ["%s"],
+				"refCondData": "cond_data1"
+			},
+			"static_rule1": {
+				"pccRuleId": "static_rule1",
+				"refCondData": "cond_data1"
+			}
+		},
+		"qosDesc": {
+			"qos_data1": {
+				"qodId": "qos_data1",
+				"5qi": 5,
+				"maxbrUl": "200000",
+				"maxbrDl": "500000",
+				"gbrUl": "100000",
+				"gbrDl": "250000"
+			}
+		},
+		"chgDecs": {
+			"chg_data1": {
+				"chgId": "chg_data1",
+				"online": true,
+				"ratingGroup": 1,
+				"serviceId": 12
+			}
+		},
+		"traffContDecs": {
+			"tc_data1": {
+				"tcId": "tc_data1",
+				"redirectInfo": {
+					"redirectEnabled": true,
+					"redirectAddressType": "URL",
+					"redirectServerAddress": "https://redirect.example.com/tc"
+				}
+			}
+		},
+		"umDecs": {
+			"%s": {
+				"umId": "%s",
+				"volumeThreshold": 4000000,
+				"volumeThresholdUplink": 1500000,
+				"volumeThresholdDownlink": 3500000
+			}
+		},
+		"conds": {
+			"cond_data1": {
+				"condId": "cond_data1",
+				"activationTime": "2021-10-22T12:42:31Z",
+				"deactivationTime": "2021-10-22T14:42:31Z"
+			}
+		},
+		"policyCtrlReqTriggers": ["RE_TIMEOUT"],
+		"revalidationTime": "2021-10-22T14:42:31Z",
+		"online": true
+	}`
+	return fmt.Sprintf(policyDecisionStr, RULE_ID1, UM_DATA1, UM_DATA1, UM_DATA1)
+}
+
+func createPolicyUpdateAnswer() string {
+	policyDecisionStr := `{
+		"pccRules": {
+			"rule1": {
+				"pccRuleId": "%s",
+				"flowInfos": [{
+					"flowDescription": "permit in ip from 0.0.0.0/0 to 4.2.2.4"
+				}],
+				"precedence": 1,
+				"refUmData": ["%s"]
+			}
+		},
+		"umDecs": {
+			"%s": {
+				"umId": "%s",
+				"volumeThreshold": 4000000,
+				"volumeThresholdUplink": 1500000,
+				"volumeThresholdDownlink": 3500000
+			}
+		}
+		"policyCtrlReqTriggers": ["RE_TIMEOUT"],
+		"revalidationTime": "2021-10-22T14:42:31Z",
+		"online": true
+	}`
+	return fmt.Sprintf(policyDecisionStr, RULE_ID1, UM_DATA1, UM_DATA1, UM_DATA1)
+}
+
+func createAccuUsageReport() string {
+	usageReport := `{
+		"refUmIds": "%s",
+		"volUsageUplink": %d,
+		"volUsageDownlink": %d,
+		"volUsage": %d
+	}`
+	return fmt.Sprintf(usageReport, UM_DATA1, UsageTx1, UsageRx1, UsageTotal1)
+}
```

Comment:
Note the mock driven tests was thought as a quick way to start running cwag integ test. 
I think it would be more valuable if we could build a working PCF, were we can load subscribers/profiles/rules though grpc calls instead of just a mock one.

Reply:
GRPC based subscriber/rule additions similar to mock PCRF are supported. Will have to add a unit test for that.